bgpd/bgp_packet.c

```diff
@@ -1005,8 +1005,16 @@ static int bgp_collision_detect(struct peer *new, struct in_addr remote_id)
 					return -1;
 				}
 			else {
-				if (ntohl(peer->local_id.s_addr) ==
-				    ntohl(remote_id.s_addr))
+				/* rfc6286:
+				 * If the BGP Identifiers of the peers
+				 * involved in the connection collision
+				 * are identical, then the connection
+				 * initiated by the BGP speaker with the
+				 * larger AS number is preserved.
+				 */
+				if (ntohl(peer->local_id.s_addr)
+					    == ntohl(remote_id.s_addr)
+				    && peer->local_as < peer->as)
```

Comment:
Is this feature an option or must have? If option we need to control it to enable disable it.

Reply:
Frankly speaking it's not a feature ;-) It's just how RFC6286 defines:
>These
   revisions to the base BGP specification do not introduce any backward
   compatibility issues.